protocols/mdns/src/query.rs

```diff
@@ -218,9 +208,9 @@ impl MdnsPeer {
     pub fn new(
         packet: &Packet<'_>,
         record_value: String,
-        my_peer_id: PeerId,
         ttl: u32,
-    ) -> MdnsPeer {
+    ) -> Result<MdnsPeer, String> {
```

Comment:
@mxinden @MarcoPolo should this return a `Result<MdnsPeer, _>` or an `Option<MdnsPeer>`? I feel that the first makes it clearer.

Reply:
I think this is a little tricky, but I think it comes down to two things:
1. Is the returned error helpful or actionable? Can it be propagated to the user, or can the caller do something with that error that is useful?
2. Is the returned error correct (This is actually a corollary to the above)

For 1, I don't think the caller can do anything different. Either it can get a peer from the record+packet, or it can't. The error won't change anything. This is true for `MdnsPacket` too. In some other system you can imagine an error leading to a retry, or a user prompt, or some error specific remediation, but not here.

For 2, I don't think the error returned is always correct. There are many branches that return `None` in the logic below, and they aren't all related to a missing peer id. So this error could actually be confusing if the error was actually that there weren't any `TXT` fields (for example).

So my thoughts are that this should be an `Option<...>` like `MdnsPacket`.

Reply:
Resolved in 7dc6b1f